### ISSUE TITLE
S1128 FP: Collection Initializer Extension Methods

### ISSUE THREAD

**Issue by dbremner** [SonarSource/sonar-dotnet#3065]
### Description

C# 6 added support for writing [collection initializers](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers) that are extension methods.

A S1128 warning is shown when a source file uses collection initializer extension methods even if the using is required.

### Repro steps

The attached [S1128FalsePositive.zip](https://github.com/SonarSource/sonar-dotnet/files/4092874/S1128FalsePositive.zip) contains a minimal .NET Framework project that will reproduce this issue.

1. Extract attached ZIP file.

2. Open S1128FalsePositive.sln with Visual Studio when the SonarLint extension is enabled.

3. You will see a S1128 warning for the **S1128FalsePositive.Extensions** using in Program.cs.

4. Remove the **S1128FalsePositive.Extensions** using and observe that the project will fail to build.

### Expected behavior

No S1128 warning is shown for this case.

### Actual behavior

A S1128 warning is displayed even though the using is required.

### Known workarounds

Suppressing the warning, modifying the source file, or disabling the SonarLint extension.

### Related information

* SonarLint - 4.16.0.14364
* SonarC# Version - is this the version of SonarAnalyzer.dll? If so, it is 8.1.0.13383.
* Visual Studio Version - 2017 Enterprise 15.9.19

**Comment by pavel-mikula-sonarsource** [comment]
Hi @dbremner, thank you for reporting this.

Here's minimal reproducible example for this case:
```
using System.Collections.Generic;

namespace S1128FalsePositive.Extensions
{
    public static class ListExtensions
    {
        public static void Add(this List<string> list, string firstName, string lastName)
        {
            list.Add(firstName + " " + lastName);
        }
    }
}

namespace S1128FalsePositive
{
    using S1128FalsePositive.Extensions;

    internal static class Program
    {
        private static void Main(string[] args)
        {
            var list = new List<string>
            {
                { "John", "Smith" },
            };
        }
    }
}
```
